server/lib/set-headers-for-date-temporal-context.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const assert = require('assert');
+
+const { getUtcStartOfDayTs } = require('matrix-public-archive-shared/lib/timestamp-utilities');
+
+// `X-Date-Temporal-Context` indicates the temporal context of the content, whether it
+// is related to past, present, or future *day*. This is useful for caching purposes so
+// you can heavily cache past content, but not present/future.
+function setHeadersForDateTemporalContext({ res, nowTs, comparedToUrlDate: { yyyy, mm, dd } }) {
+  assert(res);
+  assert(nowTs);
+  assert(yyyy);
+  assert(mm);
+  assert(dd);
+
+  // We use the start of the UTC day so we can compare apples to apples and see whether
+  // yyyy-mm-dd is the past/present/future day.
+  const startOfTodayTs = getUtcStartOfDayTs(nowTs);
+  const compareTs = Date.UTC(yyyy, mm, dd);
+
+  let temporalContext = 'present';
+  if (compareTs < startOfTodayTs) {
+    temporalContext = 'past';
+  } else if (compareTs > startOfTodayTs) {
+    temporalContext = 'future';
+  }
+
+  res.set('X-Date-Temporal-Context', temporalContext);
```

Comment:
Currently using:

`X-Date-Temporal-Context`: Indicates the temporal context of the content, whether it is related to past, present or future day.

Better names?

 - `X-Date-Context`
 - `X-Date-Status`
 - `X-Date-Comparison`: Indicates whether the date in the URL is before, after, or the same as the current date.
 - `X-Date-Relativity`: Indicates the relationship of the content to the current date, whether it is from the past, present or future.
 - `X-Date-Categorization`: Indicates the categorization of the content based on its date, whether it is classified as past, present or future content.